### ISSUE TITLE
node.data can be undefined?

### ISSUE THREAD

**Issue by SamyPesse** [ianstormtaylor/slate#194]
I'm working on building a POC of Slate in our application.

When switching a header into a code block, an exception is thrown by our code block node component (which is using `node.data.get`): `Uncaught TypeError: Cannot read property 'get' of undefined`

It looks like the data property can be undefined. Shouldn't it default to an empty `Data()`?

![screen shot 2016-07-28 at 18 22 22](https://cloud.githubusercontent.com/assets/845425/17220724/cc5953e0-54f0-11e6-840b-5cced9c7eeff.png)

**Comment by SamyPesse** [comment]
The issue is that the `setBlock` transform replace the data if we call it with `data: undefined`:

```js
state.transform()
    .setBlock({ type: 'blockType', data: undefined })
    .apply()
```

It will break the `data` property of the node.

I'll make a PR to make it impossible to replace the `data` property by something else than a `Data`.

**Comment by ianstormtaylor** [comment]
Fixed in `0.8.1`, thanks for the PR!
